Add Template for Docker & Parse

felixrieseberg opened the pull request:
This PR adds a template for deploying Parse on Azure, using Docker.

azurecla commented:
Hi __@felixrieseberg__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (feriese). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

felixrieseberg commented:
Paging Dr @sedouard ;)

singhkay commented on docker-parse/README.md:
```diff
@@ -0,0 +1,11 @@
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template allows you to deploy an Ubuntu Server 15.04 VM with Docker (using the [Docker Extension](https://github.com/Azure/azure-docker-extension))
```
Use the 15.10 image. 15.04 is EOL and will be removed from the image repository soon causing this template to break

singhkay commented on docker-parse/azuredeploy.json:
```diff
@@ -0,0 +1,273 @@
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "newStorageAccountName": {
```
Eliminate this to avoid failures. See #944

singhkay commented on docker-parse/azuredeploy.json:
```diff
@@ -0,0 +1,273 @@
+      "metadata": {
+        "description": "The location where the Virtual Machine will be deployed."
+      },
+      "allowedValues": [
```
Since VMs are now available everywhere, it's better to leave the allowed values out

singhkay commented on docker-parse/azuredeploy.json:
```diff
@@ -0,0 +1,273 @@
+        "description": "Password for the Virtual Machine."
+      }
+    },
+    "dnsNameForPublicIP": {
```
Call this a prefix. See #966

singhkay commented on docker-parse/azuredeploy.json:
```diff
@@ -0,0 +1,273 @@
+  "variables": {
+    "imagePublisher": "Canonical",
+    "imageOffer": "UbuntuServer",
+    "ubuntuOSVersion": "15.04",
```
15.10 or LTS to make this template work for a longer time

singhkay commented on docker-parse/azuredeploy.json:
```diff
@@ -0,0 +1,273 @@
+    "publicIPAddressType": "Dynamic",
+    "vmStorageAccountContainerName": "vhds",
+    "vmName": "MyDockerVM",
+    "vmSize": "Standard_D1",
```
Is the migrated app going to fine with a D1? Maybe offer this as a parameter with some more sizes as allowed values?

singhkay commented on docker-parse/azuredeploy.json:
```diff
@@ -0,0 +1,273 @@
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('newStorageAccountName')]",
+      "apiVersion": "2015-05-01-preview",
```
make apiversion a variable and reuse across templates. Also use the GA API 2015-06-15

singhkay commented on docker-parse/azuredeploy.parameters.json:
```diff
@@ -0,0 +1,30 @@
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "newStorageAccountName": {
```
See placeholder parameters for passing CI https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE#travis-ci

felixrieseberg commented:
@singhkay: Thanks for the speedy feedback. I addressed all concerns, I'm just waiting for Steven's Travis build to confirm that I didn't miss anything.

felixrieseberg commented:
@singhkay: All issues fixed, tests passed!

singhkay commented:
lgtm, merged! This was pretty quick given the parse announcement yesterday :smile:

felixrieseberg commented:
http://www.felixrieseberg.com/running-your-own-parse-with-azure-and-docker/ ;-)

felixrieseberg commented:
@singhkay: Quick hint though, the PR is not yet merged

singhkay commented:
@felixrieseberg my bad, forgot the click. merged now
